R/data.table.R

```diff
@@ -1885,36 +1885,36 @@ as.matrix.data.table = function(x, rownames=NULL, rownames.value=NULL, ...) {
            " but should be nrow(x)==", nrow(x))
   }
   if (!is.null(rownames)) {
-    # extract that column and drop it.
+    # extract that column.
     rownames.value = x[[rownames]]
-    dm = dim(x) - 0:1
+    dm = dim(x) - 0L:1L
```

Comment:
FWIW this is a case where you can save your pinky and skip the `L` 😅 

`identical(0:1, 0L:1L)`, I'm not sure whether it's handled on compilation or whether `:` does it, but they're essentially the same computationally as well:

```
> microbenchmark::microbenchmark(times = 1e5, 0L:1L, 0:1, 0L:1, 0:1L)
Unit: nanoseconds
  expr min  lq     mean median  uq     max neval cld
 0L:1L 105 123 184.9438    130 139 2110591 1e+05   a
   0:1 100 117 215.0479    123 132 4038458 1e+05   a
  0L:1 103 120 155.4638    127 136   40987 1e+05   a
  0:1L 103 120 163.0030    127 136  734152 1e+05   a
> microbenchmark::microbenchmark(times = 1e7, 0L:1L, 0:1, 0L:1, 0:1L)
Unit: nanoseconds
  expr min  lq     mean median  uq       max neval cld
 0L:1L  90 116 159.7530    125 139   5335966 1e+07   a
   0:1  85 110 179.7336    119 132 298778801 1e+07   a
  0L:1  85 113 152.6659    122 136   6443767 1e+07   a
  0:1L  86 114 186.3674    122 136 291281100 1e+07   a
```

Reply:
The same as long as we use int32 numbers

Reply:
You're right, it looks like `:` implicitly casts to integer (and is used to do some throughout the tests)